pkg/collector/corechecks/system/wincrashdetect/probe/wincrashprobe.go

```diff
@@ -11,38 +11,113 @@ import (
 	"fmt"
 	"os"
 	"path/filepath"
+	"sync"
 
 	sysconfigtypes "github.com/DataDog/datadog-agent/cmd/system-probe/config/types"
 	"github.com/DataDog/datadog-agent/pkg/util/winutil"
 	"golang.org/x/sys/windows/registry"
 )
 
+type probeState uint32
+
+const (
+	// Idle indicates that the probe is waiting for a request
+	Idle probeState = iota
+
+	// Busy indicates that the probe is currently processing a crash dump
+	Busy
+
+	// Completed indicates that the probe finished processing a crash dump.
+	Completed
+
+	// Failed indicates that the probe failed to process a crash dump.
+	Failed
+)
+
 // WinCrashProbe has no stored state.
 type WinCrashProbe struct {
+	state  probeState
+	status *WinCrashStatus
+	mu     sync.Mutex
 }
 
 // NewWinCrashProbe returns an initialized WinCrashProbe
 func NewWinCrashProbe(_ *sysconfigtypes.Config) (*WinCrashProbe, error) {
-	return &WinCrashProbe{}, nil
+	return &WinCrashProbe{
+		state:  Idle,
+		status: nil,
+	}, nil
+}
+
+// Handles crash dump parsing in a separate thread since this may take very long.
+func (p *WinCrashProbe) parseCrashDumpAsync() {
+	if p.status == nil {
+		p.state = Failed
+		return
+	}
+
+	parseCrashDump(p.status)
+
+	p.mu.Lock()
+	defer p.mu.Unlock()
+	p.state = Completed
 }
 
 // Get returns the current crash, if any
 func (p *WinCrashProbe) Get() *WinCrashStatus {
 	wcs := &WinCrashStatus{}
 
-	err := wcs.getCurrentCrashSettings()
-	if err != nil {
-		wcs.ErrString = err.Error()
-		wcs.Success = false
-		return wcs
-	}
+	// Nothing in this method should take long.
+	p.mu.Lock()
+	defer p.mu.Unlock()
+
+	switch p.state {
+	case Idle:
+		// This is a new request.
+		err := wcs.getCurrentCrashSettings()
```

Comment:
I realize this is holdover from the original code; but since we're here.
Since the settings can't change, we could do this once at init time (which would also shorten the lock time, although as you point out that time's not long anyway).

Reply:
Since the state machine will not reset, the settings will be read only once, we don't need to cache the settings.

Reply:
Actually, caching settings will simply the unit tests by shimming the registry reading part.